js_src/rdm.js

```diff
@@ -5,146 +5,973 @@ angular.module('rdmApp', [])
-   var device0 = (device >> 24) & 0xff;
-   var device1 = (device >> 16) & 0xff;
-   var device2 = (device >> 8) & 0xff;
-   var device3 = device & 0xff;
-
-   euid_bytes.push(manufacturer0 | 0xaa);
-   euid_bytes.push(manufacturer0 | 0x55);
-   euid_bytes.push(manufacturer1 | 0xaa);
-   euid_bytes.push(manufacturer1 | 0x55);
-
-   euid_bytes.push(device0 | 0xaa);
-   euid_bytes.push(device0 | 0x55);
-   euid_bytes.push(device1 | 0xaa);
-   euid_bytes.push(device1 | 0x55);
-   euid_bytes.push(device2 | 0xaa);
-   euid_bytes.push(device2 | 0x55);
-   euid_bytes.push(device3 | 0xaa);
-   euid_bytes.push(device3 | 0x55);
-
-   var checksum = euid_bytes.reduce(
-    function(previousValue, currentValue, index, array) {
-     return previousValue + currentValue;
-    }
-   );
-   var checksum0 = (checksum >> 8) & 0xff;
-   var checksum1 = checksum & 0xff;
-
-   euid_bytes.push(checksum0 | 0xaa);
-   euid_bytes.push(checksum0 | 0x55);
-   euid_bytes.push(checksum1 | 0xaa);
-   euid_bytes.push(checksum1 | 0x55);
-
-   convertor.euid = euid_bytes.map(function(i) {
-    return i.toString(16);
-   }).join(' ');
+  var parseUIDOrSetError = function(uid, uid_name) {
+   var parse_return = parserService.uidToBytes(uid);
+   if (parse_return[0]) {
+    $scope.error = 'Invalid ' + uid_name + ' UID:' + parse_return[0];
+    return '';
+   }
+   return parse_return[1];
+  };
+
+  var showHideParamData = function() {
+   $scope.properties.show_param_data = (
+       rdmHelperService.isGetSet($scope.packet.command_class) &&
+       ($scope.packet.response_type === RDM.RESPONSE_TYPE.ACK ||
+        $scope.packet.response_type === RDM.RESPONSE_TYPE.ACK_OVERFLOW));
   };
+
+  var commandClassChanged = function(new_value, old_value, scope) {
+    $scope.packet.command_class = new_value;
+    $scope.properties.is_request = rdmHelperService.isRequest(new_value);
+    showHideParamData();
+
+    var pid_names = [];
+    if (new_value === RDM.COMMAND_CLASS.DISCOVERY_COMMAND) {
+     pid_names = ['DISC_UNIQUE_BRANCH', 'DISC_MUTE', 'DISC_UN_MUTE'];
+    } else if (new_value === RDM.COMMAND_CLASS.DISCOVERY_COMMAND_RESPONSE) {
+     pid_names = ['DISC_MUTE', 'DISC_UN_MUTE'];
+    } else {
+     var discovery_pids = ['DISC_UNIQUE_BRANCH', 'DISC_MUTE', 'DISC_UN_MUTE'];
+     angular.forEach(RDM.PIDS, function(value, key) {
+       if (discovery_pids.indexOf(key) === -1) {
+        this.push(key);
+       }
+      }, pid_names);
+    }
+
+    var found_selected = false;
+    var pids = [];
+    angular.forEach(pid_names, function(key, index) {
+     found_selected |= $scope.packet.param_id === RDM.PIDS[key];
+     var label = key + ' (' + formatService.toHex(RDM.PIDS[key], 4, '0x') + ')';
+     this.push({'label': label, 'value': RDM.PIDS[key]});
+    }, pids);
+
+    pids.sort(function(a, b) {
+     return (( a.label === b.label ) ? 0 : ( ( a.label > b.label ) ? 1 : -1 ));
+    });
+
+    if (!found_selected) {
+     $scope.packet.param_id = pids[0].value;
```

Comment:
This is all still using the indent:1 jshint. When we update the config I'll do a cleanup change.